cmd/util/ledger/reporters/account_reporter.go

```diff
@@ -365,40 +371,33 @@ func (c *balanceProcessor) storageUsed(address flow.Address) (uint64, error) {
 }
 
 func (c *balanceProcessor) isDapper(address flow.Address) (bool, error) {
-	id := resourceId(address,
-		interpreter.PathValue{
-			Domain:     common.PathDomainPublic,
-			Identifier: "dapperUtilityCoinReceiver",
-		})
-
-	receiver, err := c.st.Get(id.Owner, id.Controller, id.Key, false)
+	receiver, err := c.ReadStored(address, common.PathDomainPublic, "dapperUtilityCoinReceiver")
 	if err != nil {
 		return false, fmt.Errorf("could not load dapper receiver at %s: %w", address, err)
 	}
-	return len(receiver) != 0, nil
+	return receiver != nil, nil
 }
 
 func (c *balanceProcessor) hasReceiver(address flow.Address) (bool, error) {
-	id := resourceId(address,
-		interpreter.PathValue{
-			Domain:     common.PathDomainPublic,
-			Identifier: "flowTokenReceiver",
-		})
+	receiver, err := c.ReadStored(address, common.PathDomainPublic, "flowTokenReceiver")
 
-	receiver, err := c.st.Get(id.Owner, id.Controller, id.Key, false)
 	if err != nil {
 		return false, fmt.Errorf("could not load receiver at %s: %w", address, err)
 	}
-	return len(receiver) != 0, nil
+	return receiver != nil, nil
 }
 
-func resourceId(address flow.Address, path interpreter.PathValue) flow.RegisterID {
-	// Copied logic from interpreter.storageKey(path)
-	key := fmt.Sprintf("%s\x1F%s", path.Domain.Identifier(), path.Identifier)
-
-	return flow.RegisterID{
-		Owner:      string(address.Bytes()),
-		Controller: "",
-		Key:        key,
+func (c *balanceProcessor) ReadStored(address flow.Address, domain common.PathDomain, id string) (cadence.Value, error) {
+	addr, err := common.HexToAddress(address.Hex())
```

Comment:
I think we can do this more efficient using bytes than conversion to hex and decode from hex,